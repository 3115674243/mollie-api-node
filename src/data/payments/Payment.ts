import { PaymentData } from './data';
import Chargeback, { injectPrototypes as injectChargebackPrototypes } from '../chargebacks/Chargeback';
import Refund, { injectPrototypes as injectRefundPrototypes } from '../refunds/Refund';
import Seal from '../../types/Seal';
import paymentHelpers from './helpers';

type Payment = Seal<
  PaymentData & {
    _embedded?: {
      refunds?: Omit<Refund, '_embedded'>[];
      chargebacks?: Omit<Chargeback, '_embedded'>[];
    };
  },
  typeof paymentHelpers
>;

export default Payment;

export function injectPrototypes(input: PaymentData): Payment {
  let _embedded: Payment['_embedded'];
  if (input._embedded != undefined) {
    _embedded = {};
    if (input._embedded.chargebacks != undefined) {
      _embedded.chargebacks = input._embedded.chargebacks.map(injectChargebackPrototypes);
    }
    if (input._embedded.refunds != undefined) {
      _embedded.refunds = input._embedded.refunds.map(injectRefundPrototypes);
    }
  }
  return Object.assign(Object.create(paymentHelpers), input, { _embedded });
}
