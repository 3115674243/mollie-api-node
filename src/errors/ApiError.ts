import { MollieApiError, MollieApiErrorLinks, Url } from '../data/global';
import { cloneDeep, get, has } from 'lodash';

/**
 * @since 3.0.0
 */
export default class ApiError extends Error {
  public constructor(message: string, protected title?: string, protected status?: number, protected field?: string, protected links?: MollieApiErrorLinks) {
    super(message);
    this.name = 'ApiError';
  }

  /**
   * Get the error message
   *
   * @returns The error message
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getMessage(): string {
    return this.message;
  }

  /**
   * Get the field name that contains an error
   *
   * @returns The error field
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getField(): string | undefined {
    return this.field;
  }

  /**
   * Get the API status code
   *
   * @returns The status code
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getStatusCode(): number | undefined {
    return this.status;
  }

  /**
   * Get the documentation URL
   *
   * @returns The documentation URL
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getDocumentationUrl(): string {
    return this.getUrl('documentation');
  }

  /**
   * Get the dashboard URL
   *
   * @returns The dashboard URL
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getDashboardUrl(): string {
    return this.getUrl('dashboard');
  }

  /**
   * Check if the link exists
   *
   * @param key - Link name
   *
   * @returns Whether the link exists
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public hasLink(key: string): boolean {
    return has(this.links, key);
  }

  /***
   * Retrieve a link by name
   *
   * @param key - The link name
   *
   * @returns A link to the resource
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getLink(key: string): Url {
    return get(this.links, key);
  }

  /**
   *
   * @param {string} key
   *
   * @returns {string}
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getUrl(key: string): string {
    return get(this.getLink(key), 'href');
  }

  public toString(): string {
    return `ApiError: ${this.message}`;
  }

  /**
   * Create an `ApiError` from a raw error format
   *
   * @param error - A raw Mollie API error
   *
   * @returns A new `ApiError`
   *
   * @since 3.0.0
   */
  public static createFromResponse(error: MollieApiError): ApiError {
    return new ApiError(get(error, 'data.detail'), get(error, 'data.title'), get(error, 'data.status'), get(error, 'data.field'), cloneDeep(get(error, 'data._links')));
  }
}
