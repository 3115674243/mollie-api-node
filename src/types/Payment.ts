declare namespace Mollie {
  interface IPayment {
    resource: string;
    id: string;
    mode: ApiMode;
    createdAt: string;
    status: Payment.Status;
    isCancelable: boolean;
    authorizedAt?: string;
    paidAt?: string;
    canceledAt?: string;
    expiresAt: string;
    expiredAt?: string;
    failedAt?: string;
    amount: IAmount;
    amountRefunded?: IAmount;
    amountRemaining?: IAmount;
    amountCaptured?: IAmount;
    description: string;
    redirectUrl: string | null;
    webhookUrl?: string;
    method: Method;
    metadata: any;
    locale: Locale;
    countryCode?: string;
    profileId: string;
    settlementAmount?: IAmount;
    settlementId?: string;
    customerId?: string;
    sequenceType: string;
    mandateId?: string;
    subscriptionId?: string;
    orderId?: string;
    details?: any;
    applicationFee?: {
      amount: IAmount;
      description: string;
    };
    _links: ILinks;
  }

  namespace Payment {
    enum Status {
      open = 'open',
      canceled = 'canceled',
      pending = 'pending',
      authorized = 'authorized',
      expired = 'expired',
      failed = 'failed',
      paid = 'paid',
    }

    type Include = 'details.qrCode';

    enum Embed {
      refunds = 'refunds',
      chargebacks = 'chargebacks',
    }

    namespace Params {
      interface ICreate {
        amount: IAmount;
        description: string;
        redirectUrl?: string;
        webhookUrl?: string;
        locale?: Locale;
        method?: Method;
        metadata?: any;
        sequenceType?: string;
        customerId?: string;
        mandateId?: string;
        billingEmail?: string;
        billingAddress?: IAddress;
        shippingAddress?: IAddress;
        issuer: GiftcardIssuer | IdealIssuer | KbcIssuer;
        dueDate?: string;
        customerReference?: string;
        consumerName?: string;
        consumerAccount?: string;
        details?: any;

        // Access token parameters
        profileId?: string;
        testmode?: boolean;
        applicationFee?: IAmount;
      }

      interface IGet {
        includes?: Include;
        embed?: Array<Embed>; // TODO: add support for embeds

        // Access token parameters
        testmode?: boolean;
      }

      interface IList {}

      interface IUpdate extends Partial<IPayment> {}

      interface ICancel {
        // Access token parameters
        testmode?: boolean;
      }
    }

    namespace Callback {
      type Create = (err: any, payment: IPayment) => void;
      type Get = (err: any, payment: IPayment) => void;
      type List = (err: any, payments: Mollie.List<IPayment>) => void;
      type Update = (err: any, payment: IPayment) => void;
      type Cancel = (err: any, payment: IPayment) => void;
    }
  }
}
