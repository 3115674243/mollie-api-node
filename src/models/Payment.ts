import Model from '../model';
import { IPayment, PaymentStatus } from '../types/payment';
import { ApiMode, IAmount, ILinks, Locale, PaymentMethod } from '../types/global';

/**
 * The `Payment` model
 */
export default class Payment extends Model implements IPayment {
  public static readonly resourcePrefix = 'tr_';
  public resource: string;
  public id: string;

  public mode: ApiMode;
  public createdAt: string;
  public status: PaymentStatus;
  public isCancelable: boolean;
  public authorizedAt?: string;
  public paidAt?: string;
  public canceledAt?: string;
  public expiresAt: string;
  public expiredAt?: string;
  public failedAt?: string;
  public amount: IAmount;
  public amountRefunded?: IAmount;
  public amountRemaining?: IAmount;
  public amountCaptured?: IAmount;
  public description: string;
  public redirectUrl: string | null;
  public webhookUrl?: string;
  public method: PaymentMethod;
  public metadata: any;
  public locale: Locale;
  public countryCode?: string;
  public profileId: string;
  public settlementAmount?: IAmount;
  public settlementId?: string;
  public customerId?: string;
  public sequenceType: string;
  public mandateId?: string;
  public subscriptionId?: string;
  public orderId?: string;
  public details?: any;
  public applicationFee?: {
    amount: IAmount;
    description: string;
  };
  public _links: ILinks;

  constructor(props?: Partial<IPayment>) {
    super(props);

    const defaults: Partial<IPayment> = {
      resource: 'payment',
      id: null,
      mode: null,
      createdAt: null,
      status: null,
      isCancelable: null,
      paidAt: null,
      canceledAt: null,
      expiresAt: null,
      expiredAt: null,
      failedAt: null,
      amount: {
        value: null,
        currency: null,
      },
      amountRefunded: null,
      amountRemaining: null,
      description: null,
      redirectUrl: null,
      webhookUrl: null,
      method: null,
      metadata: null,
      locale: null,
      countryCode: null,
      profileId: null,
      settlementAmount: null,
      settlementId: null,
      customerId: null,
      sequenceType: null,
      mandateId: null,
      subscriptionId: null,
      applicationFee: {
        amount: {
          value: null,
          currency: null,
        },
        description: null,
      },
      details: null,
      _links: {
        checkout: null,
        refunds: null,
        chargebacks: null,
        settlement: null,
        mandate: null,
        subscription: null,
        customer: null,
      },
    };

    Object.assign(this, defaults, props);
  }

  /**
   * If the payment is open
   *
   * @returns {boolean}
   */
  isOpen() {
    return this.status === PaymentStatus.open;
  }

  /**
   * If the payment is authorized
   *
   * @returns {boolean}
   */
  isAuthorized() {
    return this.status === PaymentStatus.authorized;
  }

  /**
   * If the payment is paid
   *
   * @returns {boolean}
   */
  isPaid() {
    return !!this.paidAt;
  }

  /**
   * If the payment is canceled
   *
   * @returns {boolean}
   */
  isCanceled() {
    return !!this.canceledAt;
  }

  /**
   * If the payment is expired
   *
   * @returns {boolean}
   */
  isExpired() {
    return !!this.expiredAt;
  }

  /**
   * If the payment is refundable
   *
   * @returns {boolean}
   *
   * @since 2.0.0-rc.2
   */
  isRefundable() {
    return this.amountRemaining !== null;
  }

  /**
   * Get the payment URL
   *
   * @returns {string|null}
   */
  getPaymentUrl() {
    return this._links && this._links.checkout && this._links.checkout.href;
  }
}
