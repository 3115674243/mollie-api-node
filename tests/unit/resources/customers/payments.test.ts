import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import CustomersPayments from '@resources/customers/payments';
import Payment from '@models/Payment';
import response from '@tests/unit/__stubs__/customers_payments.json';
import ApiError from '@errors/ApiError';

const mock = new MockAdapter(axios);

const props = {
  id: 'tr_McVBfE9Ceq',
  amount: {
    currency: 'EUR',
    value: '10.00',
  },
  description: 'Test customer payment of â‚¬10.00',
  customerId: 'cst_c24gk2t3G6',
  redirectUrl: 'https://www.example.org/',
};

describe('customers_payments', () => {
  let customersPayments: CustomersPayments;
  beforeEach(() => {
    customersPayments = new CustomersPayments(axios.create());
  });

  it('should have a resource name and model', () => {
    expect(CustomersPayments.resource).toBe('customers_payments');
    expect(CustomersPayments.model).toBe(Payment);
  });

  describe('.create()', () => {
    mock.onPost(`/customers/${props.customerId}/payments`).reply(200, response._embedded.payments[0]);

    it('should return a payment instance', () =>
      customersPayments.create(props).then(result => {
        expect(result).toBeInstanceOf(Payment);
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      customersPayments.create(props, (err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Payment);
        expect(result).toMatchSnapshot();
        done();
      });
    });

    it('should work with withParent', done => {
      customersPayments
        .withParent({
          resource: 'customer',
          id: props.customerId,
        })
        .create(props, (err, result) => {
          expect(err).toBeNull();
          expect(result).toBeInstanceOf(Payment);
          expect(result).toMatchSnapshot();
          done();
        });
    });
  });

  describe('.list()', () => {
    const error = { detail: 'The customer id is invalid' };
    mock.onGet(`/customers/${props.customerId}/payments`).reply(200, response);

    it('should return a list of all customer payments', done => {
      customersPayments
        .list({ customerId: props.customerId })
        .then(result => {
          expect(result).toBeInstanceOf(Array);
          expect(result).toHaveProperty('links');
          expect(result).toMatchSnapshot();
          done();
        })
        .catch(err => expect(err).toBeUndefined());
    });

    it('should throw an error if "customerId" is not set', done => {
      customersPayments
        .list(undefined)
        .then(() => {
          throw Error('Should error out');
        })
        .catch(err => {
          expect(err).toBeInstanceOf(ApiError);
          expect(err.getMessage()).toEqual(error.detail);
          done();
        });
    });

    it('should work with a callback', done => {
      customersPayments.list(props, (err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
        done();
      });
    });

    /**
     * @deprecated 3.0.0
     */
    it('should work with .withParent()', done => {
      customersPayments
        .withParent({
          resource: 'customer',
          id: props.customerId,
        })
        .list((err, result) => {
          expect(err).toBeNull();
          expect(result).toBeInstanceOf(Array);
          expect(result).toHaveProperty('links');
          expect(result).toMatchSnapshot();
          done();
        });
    });

    /**
     * @deprecated 3.0.0
     */
    it('should work with a Promise and with .withParent()', done => {
      customersPayments
        .withParent({
          resource: 'customer',
          id: props.customerId,
        })
        .list(undefined)
        .then(result => {
          expect(result).toBeInstanceOf(Array);
          expect(result).toHaveProperty('links');
          expect(result).toMatchSnapshot();
          done();
        })
        .catch(err => {
          expect(err).toBeNull();
          done();
        });
    });
  });
});
