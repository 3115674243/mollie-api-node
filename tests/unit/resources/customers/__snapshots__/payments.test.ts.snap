// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customers_payments .create() should return a payment instance 1`] = `
Payment {
  "_embedded": null,
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "amountCaptured": null,
  "amountRefunded": undefined,
  "amountRemaining": undefined,
  "applicationFee": undefined,
  "canceledAt": undefined,
  "countryCode": undefined,
  "createdAt": "2018-02-12T11:58:35.0Z",
  "customerId": "cst_kEn1PlbGa",
  "description": "test",
  "details": null,
  "expiredAt": undefined,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "failedAt": undefined,
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "locale": undefined,
  "mandateId": undefined,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "orderId": null,
  "paidAt": undefined,
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "sequenceType": undefined,
  "settlementAmount": undefined,
  "settlementId": undefined,
  "status": "open",
  "subscriptionId": undefined,
  "webhookUrl": undefined,
}
`;

exports[`customers_payments .create() should work with a callback 1`] = `
Payment {
  "_embedded": null,
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "amountCaptured": null,
  "amountRefunded": undefined,
  "amountRemaining": undefined,
  "applicationFee": undefined,
  "canceledAt": undefined,
  "countryCode": undefined,
  "createdAt": "2018-02-12T11:58:35.0Z",
  "customerId": "cst_kEn1PlbGa",
  "description": "test",
  "details": null,
  "expiredAt": undefined,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "failedAt": undefined,
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "locale": undefined,
  "mandateId": undefined,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "orderId": null,
  "paidAt": undefined,
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "sequenceType": undefined,
  "settlementAmount": undefined,
  "settlementId": undefined,
  "status": "open",
  "subscriptionId": undefined,
  "webhookUrl": undefined,
}
`;

exports[`customers_payments .create() should work with withParent 1`] = `
Payment {
  "_embedded": null,
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "amountCaptured": null,
  "amountRefunded": undefined,
  "amountRemaining": undefined,
  "applicationFee": undefined,
  "canceledAt": undefined,
  "countryCode": undefined,
  "createdAt": "2018-02-12T11:58:35.0Z",
  "customerId": "cst_kEn1PlbGa",
  "description": "test",
  "details": null,
  "expiredAt": undefined,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "failedAt": undefined,
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "locale": undefined,
  "mandateId": undefined,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "orderId": null,
  "paidAt": undefined,
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "sequenceType": undefined,
  "settlementAmount": undefined,
  "settlementId": undefined,
  "status": "open",
  "subscriptionId": undefined,
  "webhookUrl": undefined,
}
`;

exports[`customers_payments .list() should return a list of all customer payments 1`] = `
List [
  Payment {
    "_embedded": null,
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "amountCaptured": null,
    "amountRefunded": undefined,
    "amountRemaining": undefined,
    "applicationFee": undefined,
    "canceledAt": undefined,
    "countryCode": undefined,
    "createdAt": "2018-02-12T11:58:35.0Z",
    "customerId": "cst_kEn1PlbGa",
    "description": "test",
    "details": null,
    "expiredAt": undefined,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "failedAt": undefined,
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "locale": undefined,
    "mandateId": undefined,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "orderId": null,
    "paidAt": undefined,
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "sequenceType": undefined,
    "settlementAmount": undefined,
    "settlementId": undefined,
    "status": "open",
    "subscriptionId": undefined,
    "webhookUrl": undefined,
  },
]
`;

exports[`customers_payments .list() should work with .withParent() 1`] = `
List [
  Payment {
    "_embedded": null,
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "amountCaptured": null,
    "amountRefunded": undefined,
    "amountRemaining": undefined,
    "applicationFee": undefined,
    "canceledAt": undefined,
    "countryCode": undefined,
    "createdAt": "2018-02-12T11:58:35.0Z",
    "customerId": "cst_kEn1PlbGa",
    "description": "test",
    "details": null,
    "expiredAt": undefined,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "failedAt": undefined,
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "locale": undefined,
    "mandateId": undefined,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "orderId": null,
    "paidAt": undefined,
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "sequenceType": undefined,
    "settlementAmount": undefined,
    "settlementId": undefined,
    "status": "open",
    "subscriptionId": undefined,
    "webhookUrl": undefined,
  },
]
`;

exports[`customers_payments .list() should work with a Promise and with .withParent() 1`] = `
List [
  Payment {
    "_embedded": null,
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "amountCaptured": null,
    "amountRefunded": undefined,
    "amountRemaining": undefined,
    "applicationFee": undefined,
    "canceledAt": undefined,
    "countryCode": undefined,
    "createdAt": "2018-02-12T11:58:35.0Z",
    "customerId": "cst_kEn1PlbGa",
    "description": "test",
    "details": null,
    "expiredAt": undefined,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "failedAt": undefined,
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "locale": undefined,
    "mandateId": undefined,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "orderId": null,
    "paidAt": undefined,
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "sequenceType": undefined,
    "settlementAmount": undefined,
    "settlementId": undefined,
    "status": "open",
    "subscriptionId": undefined,
    "webhookUrl": undefined,
  },
]
`;

exports[`customers_payments .list() should work with a callback 1`] = `
List [
  Payment {
    "_embedded": null,
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "amountCaptured": null,
    "amountRefunded": undefined,
    "amountRemaining": undefined,
    "applicationFee": undefined,
    "canceledAt": undefined,
    "countryCode": undefined,
    "createdAt": "2018-02-12T11:58:35.0Z",
    "customerId": "cst_kEn1PlbGa",
    "description": "test",
    "details": null,
    "expiredAt": undefined,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "failedAt": undefined,
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "locale": undefined,
    "mandateId": undefined,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "orderId": null,
    "paidAt": undefined,
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "sequenceType": undefined,
    "settlementAmount": undefined,
    "settlementId": undefined,
    "status": "open",
    "subscriptionId": undefined,
    "webhookUrl": undefined,
  },
]
`;
